wordpress-mobile/gutenberg-mobile
# [UI tests] Run canaries on a PR and optionally full suite

In this PR I added a couple of options to make test runs more stable. 

1. **Run only a subset of tests (canaries) on every PR** - this will shorten execution time for UI tests jobs in CircleCI. All [tests marked with `@canary`](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2137/files#diff-63d46a3d5d8db0e8f1c48b497cd83222R19) will be part of these. Name of these jobs are _Test Android on Device - Canaries_ and _Test iOS on Device - Canaries_ and they are mandatory for every PR. I marked three tests with the `@canary` and we can change that and choose other tests if needed. @hypest once this PR is approved and ready to merge, we should change required jobs for `develop` branch - canaries should be marked as required and `Test Android/iOS on Device` should be unchecked. 
2. **Optionally run the full suite of UI tests** - If necessary, you can run the full suite. A comment will be posted on a PR, see [example](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2137#issuecomment-611176621) below, which will lead to CircleCI workflow where you can start Android and iOS jobs. I did this with [peril-settings](https://github.com/Automattic/peril-settings/pull/43/files), similar to what we are using in other mobile projects. **EDIT:** Full suite will run every day against `develop`, per schedule https://github.com/wordpress-mobile/gutenberg-mobile/pull/2115
3. **Increase CircleCI timeout** - In the past, we were seeing error `Too long with no output (exceeded 10m0s): context deadline exceeded`. This is happening because some tests are taking too long (more than 10m) without any output during execution, and CircleCI job will fail. I [increased the timeout to 20m](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2137/files#diff-1d37e48f9ceff6d8030570cd36286a61R88) to reduce noise. 

To test:
- Look at the jobs in this PR, compare times. 
- Optionally, add some dummy commit to trigger jobs. Open the link from the comment and trigger full suite jobs. Does it work as expected? 

PR submission checklist:
